### PR TITLE
A test for the case where the dictionary in word mode is too short

### PR DESCRIPTION
**Pull request by julianuu**

### CODE REVIEW

**Comment by The-Compiler** [comment]
Congrats on your first PR! :wink:

I'll add some line comments - you can simply push more commits to the existing PR to update it.

File: tests/end2end/data/hints/short_dict.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+
+<!-- target: hello.txt -->
```
Comment (The-Compiler) [inline tests/end2end/data/hints/short_dict.html]: You can remove that comment - it's only needed for the files in `data/hints/html` which are handled automatically (where `test_hints` verifies the hint goes to the target given in the comment).

File: tests/end2end/data/hints/short_dict.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+<html>
+    <head>
+        <meta charset="utf-8">
+	<title>Many links</title>
```
Comment (The-Compiler) [inline tests/end2end/data/hints/short_dict.html]: nitpick: please indent this by 5 more spaces, to match up with the `<meta ...>`

File: tests/end2end/test_hints_html.py [hunk 3]
```diff
@@ -113,3 +113,17 @@ def test_word_hints_issue1393(quteproc, tmpdir):
         quteproc.wait_for(message='hints: *', category='hints')
         quteproc.send_cmd(':follow-hint {}'.format(hint))
         quteproc.wait_for_load_finished('data/{}'.format(target))
+
+def test_short_dict(quteproc,tmpdir):
```
Comment (The-Compiler) [inline tests/end2end/test_hints_html.py]: Add a space after the comma please

File: tests/end2end/test_hints_html.py [hunk 4]
```diff
@@ -113,3 +113,17 @@ def test_word_hints_issue1393(quteproc, tmpdir):
...
+        a
+        b
+    """))
+    quteproc.set_setting('hints', 'mode', 'word')
+    quteproc.set_setting('hints', 'dictionary', str(dict_file))
+    quteproc.open_path('data/hints/short_dict.html')
+    quteproc.send_cmd(':hint')
+    line = quteproc.wait_for(message='Not enough words in the dictionary.')
+    line.expected = True
+    quteproc.wait_for(message='hints: *', category='hints')
```
Comment (The-Compiler) [inline tests/end2end/test_hints_html.py]: It's a bit unclear to me from looking at this line - what are you waiting for here?

File: tests/end2end/test_hints_html.py [hunk 5]
```diff
@@ -113,3 +113,17 @@ def test_word_hints_issue1393(quteproc, tmpdir):
         quteproc.wait_for(message='hints: *', category='hints')
         quteproc.send_cmd(':follow-hint {}'.format(hint))
         quteproc.wait_for_load_finished('data/{}'.format(target))
+
```
Comment (The-Compiler) [inline tests/end2end/test_hints_html.py]: Two blank lines between toplevel functions please :wink:

File: tests/end2end/test_hints_html.py (hunk 4 above)
Comment (julianuu) [inline tests/end2end/test_hints_html.py]: My idea was to test whether qutebrowser really changes to letter mode and displays hints as expected. Should that be part of the test?

Comment (The-Compiler) [inline tests/end2end/test_hints_html.py]: Ah, I see! Maybe you could even try following a hint like above?

**Comment by The-Compiler** [comment]
Thanks!

**Pull request merged**